### DISCUSSION TITLE
Dosemu 2.0pre08

### DISCUSSION THREAD

**Discussion by William0027** [dosemu2/dosemu2#1624] [Q&A]
Hi,
I am new to Linux and dosemu2. I installed dosemu2:i386 package on Ubuntu 20.04
I installed Foxpro2.6 Dos. Everything works fine, except that when I quit from Foxpro console, dosemu quit too. It didn't go back to dosemu.

ERROR: fluidsynth support not compiled in or not found:
ERROR: libfluidsynth.so.1: cannot open shared object file: No such file or directory
ERROR: alsa_midi (ALSA lib): rawmidi_hw.c:235:(snd_rawmidi_hw_open) : No such file or directory open /dev/snd/midiC0D0 failed
INFO: booting with comcom32, this is very experimental
INFO: fdpp booting, this is very experimental!
ERROR: general protection at 0xf1ea5a4f: 0

Any help would be appreciated. Thank you.

**Comment by stsp** [comment]
We do not support i386
packages any more.
This is why your package
is horribly outdated.
If you insist on an i386
build, you need to build
from git sources yourself
and get back to us.

**Reply by stsp** [reply]
And while looking into
that package, I've found
that there were no dep
on fdpp since last year,
including pre9 deb packages...
A spectacular fuck-up.

**Reply by William0027** [reply]
Ok. Thank you for your quick reply. I then installed Build 2.0pre9 version. 
Installing dosemu2 package alone is not enough, need to install fdpp package separately.
Dosemu is working fine, but I have problems with  cifs mounted folders. 
I can browse folder and subfolders contents. But when I tried to create a text file on that folder with Edit command, file is created but file size is 0. When I tried to open that file dosemu will freeze. I have to force quit dosemu. Any suggestions ?
I check on server side, the file is empty, everything I wrote is gone.
with ls -l command, I checked file permission is rwxrwxrwx, with file owner and group same as linux client user name.

**Reply by William0027** [reply]
and I notices, when I saved the file there's message at terminal console :
ERROR: MFS: failed to set xattrs for /mnt/sharename/filename.txt: Operation not supported
I don't get this message if I write to local drive in dosemu

**Reply by William0027** [reply]
change the share to NFS, still no luck. I can't even save.
ERROR: MFS: failed to set xattrs for /mnt/nfssharename/filename.txt,  unsupported !

**Reply by stsp** [reply]
> Installing dosemu2 package alone is not enough, need to install fdpp package separately.

This is what I fixed today.
So what version have you
actually installed? See
`dpkg -s dosemu2`.

I now changed the code to
not return error when xattrs
are unsupported. This may
work around your problem
so please check.

**Reply by William0027** [reply]
Package: dosemu2
Status: install ok installed
Priority: optional
Section: otherosfs
Installed-Size: 4580
Maintainer: Stas Sergeev <stsp@users.sourceforge.net>
Architecture: amd64
Version: 2.0~pre9-7384-30b4df50b+202205012047~ubuntu20.04.1
Replaces: dosemu
Depends: xfonts-utils, libbsd0 (>= 0.4.0), libc6 (>= 2.29), libreadline8 (>= 6.0), udev (>= 240), bash, coreutils, comcom32 (>= 0.1~alpha2)
Recommends: libao4 (>= 1.1.0), libasound2 (>= 1.0.16), libfluidsynth2 (>= 2.0.5), libfontconfig1 (>= 2.12.6), libgpm2 (>= 1.20.7), libieee1284-3, libjson-c4 (>= 0.13.1), libsdl2-2.0-0 (>= 2.0.10), libsdl2-image-2.0-0 (>= 2.0.2), libsdl2-ttf-2.0-0 (>= 2.0.14), libslang2 (>= 2.2.4), libslirp0 (>= 4.1.0-1), libx11-6, libxext6, ladspa-sdk, gdb, kbd, fluid-soundfont-gm, install-freedos
Suggests: vde2, valgrind, install-otherdos
Conflicts: dosemu
Conffiles:
 /etc/X11/fonts/misc/dosemu2.alias 6aac908ad7023c3b266337277db7d4a9
 /etc/dosemu/dosemu.conf c4d972f5d778db26d4bf91a68aeda21d
Description: fast and secure DOS emulator
 dosemu2 is an emulator for running DOS programs under linux.
 It can also serve as a VM to boot various DOSes.

**Reply by William0027** [reply]
I will try again and report back. Thank you.

**Reply by William0027** [reply]
Version: 2.0~pre9-7386-7928b4c17+202205021110~ubuntu20.04.1 is now installed, but problem not solved.
This time, upon saving file dosemu just crashed and exit. Before this, after saving file I still can browse the directory, only freeze when I tried to open the file again.

**Reply by William0027** [reply]
how to record the log so I can attached here ?

**Reply by William0027** [reply]
not sure if this can help :

FDPP kernel 1.6 [GIT: ] (compiled Apr 29 2022)
fdpp: plugin loaded: 1
CONF: config variable parser_version_3 set
CONF: config variable c_system set
CONF: Parsing built-in dosemu.conf file.
CONF: config variable version_3_style_used set
CONF: Parsing /etc/dosemu/dosemu.conf file.
CONF: Parsing built-in global.conf file.
CONF: mapping driver = 'auto'
debug flags: -a+cw
CONF: Disabling use of pentium timer
CONF: timer freq=18, update=54925
CONF: CPU set to 586
CONF: CPUEMU set to jit
CONF: CPU VM set to -1
CONF: CPU VM set to -1 for DPMI
CONF: 8192k bytes EMS memory
CONF: EMS-frame = 0xe000
CONF: DPMI-Server on (0x20000)
CONF: DPMI linear reserve base addr = 0xffffffff
CONF: DPMI linear reserve size = 0x8000
CONF: PM DOS API Translator on
CONF: No DJGPP NULL deref checks: on
CONF: 8192k bytes XMS memory
Warning: CONF: dosemu not running on console
CONF: time mode = 'bios'
SER: directory /var/lock namestub LCK.. binary No
MOUSE: /dev/input/mice, type 7 using internaldriver: yes, emulate3buttons: no baudrate: 0
Warning: CONF: **** Warning: floppy /dev/fd0 not accessible, disabled
CONF: fastfloppy = 1
default_drives 0
Setting up drive C, ~/.dosemu/drive_c
Creating default drive C
Added drive 0 (80): ~/.dosemu/drive_c
default_drives 1
Added drive 1 (81): /usr/share/dosemu/dosemu2-cmds-0.3
Setting up default drives from E
Added drive 2 (82): /usr/share/comcom32
Added drive 3 (83): /usr/share/dosemu2-extras/bat
CONF: IPX support off
CONF(LPT0) f: (null)   c: lpr -l  t: 20  port: 0
CONF(LPT1) f: (null)   c: lpr -P PDF  t: 20  port: 0
CONF: not allowing speaker port access
CONF: Packet Driver enabled.
CONF: NE2000 enabled.
CONF: config variable c_system unset
Command line: /usr/bin/dosemu.bin -o /home/linuxpc2/.dosemu/boot.log
Warning: Linux kernel 5.13.0; CPU speed is 1700000000 Hz
Warning: CPU-EMU speed is 1700 MHz
CONF: V86 cpu vm set to -1
CONF: DPMI cpu vm set to -1
CONF: not running on console
CONF: mostly running as USER: uid=1000 (cached 1000) gid=1000 (cached 1000)
CONF: priv operations unavailable
Warning: dosemu2-2.0pre9 is coming up on Linux version 5.13.0-40-generic #45~20.04.1-Ubuntu SMP Mon Apr 4 09:38:31 UTC 2022 x86_64
Warning: Compiled with gcc version 9.4.0Warning:  64bit
Warning: CFLAGS: -fplan9-extensions -Wall -Wstrict-prototypes -Wmissing-declarations -Wnested-externs -fms-extensions -pthread -Wno-unused-result -Wcast-qual -Wwrite-strings -Wno-address-of-packed-member -ggdb3 -fpie -g -O2 -fdebug-prefix-map=/build/dosemu2-xkNsMM/dosemu2-2.0~pre9-7386=. -fstack-protector-strong -Wformat -Werror=format-security
CONF: reserving 640Kb at 0x00000 for 'd' (Base DOS memory (first 640K))
CONF: reserving 16Kb at 0xFA000 for 'r' (Dosemu reserved area)
CONF: reserving 8Kb at 0xFE000 for 'b' (BIOS)
initial register values: fs: 0x0000  gs: 0x0000 eflags: 0x0246
initial segment bases: fs: 0x7f9b30ca6740  gs: (nil)
VID: initializing SDL plugin
CONF: reserving 128Kb at 0xA0000 for 'v' (Video memory)
Warning: Using V86 mode inside KVM
Conventional memory mapped from 0x7f9b2b89a000 to 0x40dca000
DPMI memory mapped to 0x40eda000 (reserve) and to 0x42eda000 (main)
CONF: reserving 8256Kb at 0x100000 for 'x' (Extended memory (HMA+XMS))
loading fonts for cp437
VID: initializing video sdl
VID: SDL plugin initialization completed
Registering HWRAM, type=e base=0x4b4d0000 size=0x400000
CONF: reserving 4096Kb at 0x4B4D0000 for 'e' (VGAEMU LFB)
CONF: reserving 12Kb at 0xC0000 for 'V' (VGAEMU Video BIOS)
Warning: SERIAL $Id$
Warning: Your fluidsynth is too old
DPMI: mem init, mpool is 140468224 bytes at 0x42eda000
Warning: Using DPMI inside KVM
CONF: detected layout is "us"
CONF: detected alternate layout: (null)
TIME: using 9154 usec for updating ALRM timer
CONF: freeing region for 'E' (EMS page frame)
booting with comcom32
config.boot_dos set to 400000
config.int_hooks set to 0
config.force_revect set to 0
CONF: reserving 116Kb at 0xC3000 for 'U' (Upper Memory Block (UMB, XMS 3.0))
CONF: reserving 40Kb at 0xF0000 for 'U' (Upper Memory Block (UMB, XMS 3.0))
CONF: reserving 16Kb at 0xE0000 for 'E' (EMS page frame)
CONF: reserving 16Kb at 0xE4000 for 'E' (EMS page frame)
CONF: reserving 16Kb at 0xE8000 for 'E' (EMS page frame)
CONF: reserving 16Kb at 0xEC000 for 'E' (EMS page frame)
ERROR: MFS: failed to set xattrs for /mnt/cifsmounth/tess.txt: Operation not supported
leavedos(SDL_handle_events:1425|0) called - shutting down
leavedos thread started
leavedos thread ended
coopthreads stopped

Package info:
dosemu2: installed
dosemu2	2.0~pre9-7386-7928b4c17+202205021110~ubuntu20.04.1
fdpp: installed
fdpp	1.6-1219-378e13f+202204281653~ubuntu20.04.1

**Comment by stsp** [comment]
> Version: 2.0pre9-7386-7928b4c17+202205021110ubuntu20.04.1 is now installed, but problem not solved.

This only has a fix for
bringing fdpp in, but not
for your xattrs problem.
We have daily builds.
But I restarted the build
now so that you don't
need to wait tomorrow.
Please check back in
30 minutes.

**Reply by William0027** [reply]
Noted. Thx in advance

**Comment by stsp** [comment]
In a mean time please remove
fdpp package to make sure it
will now be brought back
automatically.

**Reply by William0027** [reply]
It was brought back automatically. I removed both dosemu2 and fdpp. Only installed dosemu2 package and both were back.

**Comment by William0027** [comment]
installed new version. problem still. same error message.

**Comment by William0027** [comment]
I do a comparison with earlier installed pc, which working fine. same fstab settting, same client user name, same server

Package: fdpp
Status: install ok installed
Priority: optional
Section: otherosfs
Installed-Size: 1060
Maintainer: Stas Sergeev <stsp@users.sourceforge.net>
Architecture: amd64
Version: 1.6-1217-8b98473+202204242222~ubuntu20.04.1
Depends: libc6 (>= 2.14), libelf1 (>= 0.131), libgcc-s1 (>= 3.0), libstdc++6 (>= 5.2)
Description: 64-bit DOS core
 fdpp is a 64-bit DOS core.
 It is based on a FreeDOS kernel ported to modern C++.
Homepage: https://github.com/dosemu2/fdpp

Package: dosemu2
Versions: 
2.0~pre9-7381-0842732af+202204260253~ubuntu20.04.1 (/var/lib/apt/lists/ppa.launchpad.net_dosemu2_ppa_ubuntu_dists_focal_main_binary-amd64_Packages) (/var/lib/dpkg/status)
 Description Language: 
                 File: /var/lib/apt/lists/ppa.launchpad.net_dosemu2_ppa_ubuntu_dists_focal_main_binary-amd64_Packages
                  MD5: df9db8cf0d0e516e904d95c7ce1cf17d
 Description Language: en
                 File: /var/lib/apt/lists/ppa.launchpad.net_dosemu2_ppa_ubuntu_dists_focal_main_i18n_Translation-en
                  MD5: df9db8cf0d0e516e904d95c7ce1cf17d


Reverse Depends: 
  dosemu2:i386,dosemu2
  dosemu2-legacy,dosemu2
  dosemu2-legacy,dosemu2
Dependencies: 
2.0~pre9-7381-0842732af+202204260253~ubuntu20.04.1 - xfonts-utils (0 (null)) libbsd0 (2 0.4.0) libc6 (2 2.29) libreadline8 (2 6.0) udev (2 240) bash (0 (null)) coreutils (0 (null)) comcom32 (2 0.1~alpha2) dosemu (0 (null)) libao4 (2 1.1.0) libasound2 (2 1.0.16) libfluidsynth2 (2 2.0.5) libfontconfig1 (2 2.12.6) libgpm2 (2 1.20.7) libieee1284-3 (0 (null)) libjson-c4 (2 0.13.1) libsdl2-2.0-0 (2 2.0.10) libsdl2-image-2.0-0 (2 2.0.2) libsdl2-ttf-2.0-0 (2 2.0.14) libslang2 (2 2.2.4) libslirp0 (2 4.1.0-1) libx11-6 (0 (null)) libxext6 (0 (null)) ladspa-sdk (0 (null)) gdb (0 (null)) kbd (0 (null)) fluid-soundfont-gm (0 (null)) install-freedos (0 (null)) vde2 (0 (null)) valgrind (0 (null)) install-otherdos (0 (null)) dosemu (0 (null)) dosemu2:i386 (32 (null)) 
Provides: 
2.0~pre9-7381-0842732af+202204260253~ubuntu20.04.1 - 
Reverse Provides:

**Reply by William0027** [reply]
kernel also different. Working pc has kernel 5.4.0-109-generic, while the problem one has kernel 5.13.0-40-generic

**Reply by William0027** [reply]
also got this message on earlier installed PC , but dosemu works just fine : ERROR: MFS: failed to set xattrs for /mnt/sharename/filename.txt: Operation not supported

**Reply by stsp** [reply]
> 2.0pre9-7381-0842732af+202204260253ubuntu20.04.1

This is not a new version at all.
Please update.

**Comment by William0027** [comment]
so the difference is FDPP version and Dosemu2 version. but i suspect more on FDPP

**Reply by William0027** [reply]
I tried to get the same version of fdpp on ppa but couldn't find it

**Comment by William0027** [comment]
Okay, finally I got it working. The problem lies with kernel. In Kernel 5.4.0-109-generic dosemu2 is running fine, while in kernel 5.13.0-40-generic still stuck with freeze problem. 
This is what I did today. I fresh installed Ubuntu 18.04, tried to install dosemu2-legacy but failed. So I upgraded to 20.04. The kernel is 5.4.0-109 after upgrade ( If I do 20.04 clean install, kernel would be 5.13.0-40 ). After upgrade I just installed dosemu2 from PPA, worked like charm.
So I took another PC, which is Kernel 5.13, install latest dosemu2. Still not working :(
I will do one more try with Ubuntu 18.04 upgrade to 20.04, only install dosemu2 after upgrade. See what will happen.

**Reply by stsp** [reply]
You can just install 2 kernels
and switch them in a boot menu.
I still suspect you didn't properly
update and your dpkg-query shows
that.

**Reply by William0027** [reply]
I am new to linux so forgive me for my ignorance. I think it has to do with file permission. Somehow in kernel 5.4 dosemu is allowed to write/edit to CIFS share while in 5.13 there's problem. I repeated my installation again one more time, starting from 18.04 clean install and upgrade to 20.4, install dosemu2, worked flawlessly. So far I have repeated this 3 times, all successful.  I then tried with clean installed 5.13 kernel (second attempt), upgrade the packages and then installed dosemu2. As long as not involving CIFS share, everything is ok. But once I tried to write to CIFS share, dosemu just freezed.
Maybe there's a simple way to fix this file permission problem, but I just don't know. So for time being, my 'dumb' way is to stay with kernel 5.4. It's painstaking to upgrade from 18.04 to 20.04 though, take so long.

**Reply by William0027** [reply]
Anyway, thank you for your helps. Couldn't get dosemu running without your update/upgrade package. Greatly appreciated.

**Reply by stsp** [reply]
Maybe you need to use user_xattr
mount option to enable xattrs on
new kernels?

> dosemu just freezed

You mean completely, so that you
can't even close its window and
cursor doesn't blink?

So what dosemu2 version do you
have now?

**Reply by William0027** [reply]
Package: dosemu2
Status: install ok installed
Priority: optional
Section: otherosfs
Installed-Size: 4580
Maintainer: Stas Sergeev <stsp@users.sourceforge.net>
Architecture: amd64
Version: 2.0~pre9-7389-bfe25f0b4+202205041428~ubuntu20.04.1
Replaces: dosemu
Depends: xfonts-utils, libbsd0 (>= 0.4.0), libc6 (>= 2.29), libreadline8 (>= 6.0), udev (>= 240), bash, coreutils, fdpp, comcom32 (>= 0.1~alpha2)
Recommends: libao4 (>= 1.1.0), libasound2 (>= 1.0.16), libfluidsynth2 (>= 2.0.5), libfontconfig1 (>= 2.12.6), libgpm2 (>= 1.20.7), libieee1284-3, libjson-c4 (>= 0.13.1), libsdl2-2.0-0 (>= 2.0.10), libsdl2-image-2.0-0 (>= 2.0.2), libsdl2-ttf-2.0-0 (>= 2.0.14), libslang2 (>= 2.2.4), libslirp0 (>= 4.1.0-1), libx11-6, libxext6, ladspa-sdk, gdb, kbd, fluid-soundfont-gm, install-freedos
Suggests: vde2, valgrind, install-otherdos
Conflicts: dosemu
Conffiles:
 /etc/X11/fonts/misc/dosemu2.alias 6aac908ad7023c3b266337277db7d4a9
 /etc/dosemu/dosemu.conf c4d972f5d778db26d4bf91a68aeda21d
Description: fast and secure DOS emulator
 dosemu2 is an emulator for running DOS programs under linux.
 It can also serve as a VM to boot various DOSes.
Homepage: https://github.com/dosemu2/dosemu2

**Reply by William0027** [reply]
already add user_xattr option in fstab, still not working.

**Reply by William0027** [reply]
when it freezed, I need to click on X button to force quit dosemu.

**Reply by William0027** [reply]
When I said not working/freeze, it's when I tried to write or edit file on cifs mount. Accessing local file is fine, no problem at all.

**Reply by stsp** [reply]
Please add -D9+Dd log.

**Comment by William0027** [comment]
[debug.txt](https://github.com/dosemu2/dosemu2/files/8644532/debug.txt)

**Reply by William0027** [reply]
![Screenshot from 2022-05-07 12-11-57](https://user-images.githubusercontent.com/104744745/167239884-c73b2f39-9c37-4e46-856e-827784b13caa.png)

**Reply by stsp** [reply]
Can you give me an ssh
access to debug this?

**Reply by William0027** [reply]
I am sorry, I can't give you ssh. For me, actually I got things working with installing 18.04 then upgrade to 20.04. Very weird thing to do, but worked.
